internal/pkg/template/override/rule.go

```diff
@@ -60,7 +62,71 @@ func (r Rule) validate() error {
 }
 
 func (r Rule) parse() (nodeUpserter, error) {
-	return nil, nil
+	return r.parseRule()
+}
+
+func (r Rule) parseRule() (nodeUpserter, error) {
```

Comment:
Maybe we should start from the path. Though it is slightly more complex (same complexity), it is easier to understand?
```suggestion
func parsePath(path string) *pathSegment {
  pathSegments := strings.SplitN(r.Path, pathSegmentSeparator, 2)
  // guaranteed by `validate()`.
  current := pathSegments[0]
  if len(pathSegments) < 2 {
    return &pathSegment { current: current }
   }
  return &pathSegment {
    current: current,
    next: parsePath(pathSegments[1])
  }
}
type pathSegment {
    current string
    next *pathSegment
}
func (r Rule) parseRule() (nodeUpserter, error) {
  segment := parsePath(r.Path)
  // then construct nodes based on the parsed path.
  for segment {
    ...
    segment := segment.next
  }
}
```

Reply:
Discussed offline will keep as is.